src/components/CentralHeader/CentralHeader.tsx

```diff
@@ -14,80 +14,119 @@ import {
   Row,
   UncontrolledDropdown
 } from "reactstrap";
+import { UserRoleEnum } from "../../../generated/definitions/api/UserRole";
 
 import bootstrapItaliaImages from "../../assets/img/bootstrap-italia/sprite.svg";
 import ioLogoWhite from "../../assets/img/io-logo-white.svg";
 
+import { NonEmptyString } from "italia-ts-commons/lib/strings";
 import "./CentralHeader.css";
 
+/*
+ * Component for badge containing user role
+ */
+const UserRoleBadge = (props: { userRole: string }) => {
+  const { t } = useTranslation();
+  /**
+   * switch from user role enum to string
+   */
+  const badgeText = (userRoleEnum => {
+    switch (userRoleEnum) {
+      case UserRoleEnum.ORG_DELEGATE:
+        return t("centralHeader.userBadge.delegate");
+      case UserRoleEnum.DEVELOPER:
+        return t("centralHeader.userBadge.developer");
+      case UserRoleEnum.ORG_MANAGER:
+        return t("centralHeader.userBadge.legalRep");
+      case UserRoleEnum.ADMIN:
+        return t("centralHeader.userBadge.admin");
+    }
+  })(props.userRole);
+
+  return <Badge color="white">{badgeText}</Badge>;
+};
+
 /**
  * Props for central header
  */
-interface ICentralHeaderProps extends RouteComponentProps {
+interface IUserNameWithDropdownProps
+  extends ComponentProps<typeof UserRoleBadge>,
+    RouteComponentProps {
   userName: string;
-  userRole: string;
 }
 
 /**
- * Central header right below Slim header, containing app title and user name and user role with dropdown to go to profile page and logout
+ *  Component for user icon
  */
-export const CentralHeader = withRouter<
-  ICentralHeaderProps,
-  React.FC<ICentralHeaderProps>
->((props: ICentralHeaderProps) => {
-  /**
-   * react-i18next translation hook
-   */
-  const { t } = useTranslation();
-
-  /**
-   * create badge for user role if user role is set
-   */
-  const userRoleBadge =
-    props.userRole !== "" ? (
-      <Badge color="white">{props.userRole}</Badge>
-    ) : null;
+const UserIcon = () => {
+  return (
+    <svg className="icon icon-white">
+      <use xlinkHref={`${bootstrapItaliaImages}#it-user`} />
+    </svg>
+  );
+};
 
+/*
+ * Component for user name and role with dropdown
+ */
+const UserNameWithDropdown = withRouter((props: IUserNameWithDropdownProps) => {
+  const { t } = useTranslation();
   /**
    * function to navigate to profile
    */
   const navigateToProfile = () => props.history.push("/profile");
 
+  const userRoleBadge = NonEmptyString.decode(props.userRole).isRight() ? (
+    <UserRoleBadge userRole={props.userRole} />
+  ) : null;
+
+  const userIcon = NonEmptyString.decode(props.userName.trim()).isRight() ? (
```

Comment:
shorter: 

`const userIcon = NonEmptyString.is(props.userName.trim()) ? ...`

or 

```ts
const userIcon = NonEmptyString.decode(props.userName.trim()).fold(
   _ => null,
   _ =><UserIcon />
);
```